### PR TITLE
feat: add Flatpak permission management script

### PR DESCRIPTION
**Pull request by idnovic**
feat: harden the default flatpak permissions while keeping usability

It is not simple to understand the flatpak sandbox.
Let us try to make it simpler for the users.
This script hardens the most important settings.

But it also allows the user to add special allow rules to make sure that
-flatpak management apps
-backup apps
-custom theming
can still be used.

Why these 3?
Because we want OOTB support for flatpak UIs and backup apps.
The theming support was added because it is not simple to theme
cross ui toolkit GTK, QT. Plus flatpak and non-flatpak apps at the same time.
A user may decide to allow access to /home just because they may 
not be aware what directories are needed for custom themes.
The included config supports theme, icon, cursor and font settings OOTB.
This allows a suer if they choose so to enable theme support and start 
customising their desktop.

### CODE REVIEW

File: files/system/usr/bin/flatpak-set-perms.sh [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+# Function for System Overrides
+set_system_overrides() {
+    if [ "$(id -u)" -ne 0 ]; then
+        echo "Error: Admin rights are required to set system overrides."
+        exit 1
+    fi
+    echo "Setting system overrides..."
+    set_override "system"  # Call set_override with 'system'
```
Comment (RoyalOughtness) [inline files/system/usr/bin/flatpak-set-perms.sh:48]: Please review this issue https://github.com/secureblue/secureblue/issues/125

A good amount of this PR is out of scope

File: files/system/usr/bin/flatpak-set-perms.sh [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    local scope="$1"  # 'user' or 'system'
+    local flatpak_command="flatpak override --$scope"
+    
+    # the base override itself, we use the same override for system and user
+    # https://docs.flatpak.org/en/latest/sandbox-permissions.html
+    # we disallow access to filesystem host and home, allow partly access to host read-only
+    # we prefer the wayland socket and only allow x11 as fallback
+    # we disallow access to IPC and Devel
+    # we disallow bus socket access
+    $flatpak_command \
```
Comment (RoyalOughtness) [inline files/system/usr/bin/flatpak-set-perms.sh:34]: as already discussed on discord, users who don't know to check their app permissions are the same users who would blindly turn permissions on if something isn't working. As such, blanket globally disabling permissions isn't a good approach.

File: files/system/usr/bin/flatpak-set-perms.sh [hunk 3]
```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env bash
```
Comment (RoyalOughtness) [inline files/system/usr/bin/flatpak-set-perms.sh:1]: as a general rule for working in FOSS projects, it's good to get buy-in from the project before working on features so you don't invest time on changes that won't be accepted

**Review by RoyalOughtness — CHANGES_REQUESTED** [review]

**Comment by idnovic** [comment]
I created most of the script for my image.
If this is out of scope that is alright, too.

I wanted to suggest to leave the user the option
to use this if they choose do to that.
